lib/Basics/memory-map-posix.cpp

```diff
@@ -76,28 +76,45 @@ int TRI_FlushMMFile(int fileDescriptor, void* startingAddress,
 
   int res = msync(startingAddress, numOfBytesToFlush, flags);
 
-#ifdef __APPLE__
-  if (res == TRI_ERROR_NO_ERROR) {
-    res = fcntl(fileDescriptor, F_FULLFSYNC, 0);
-  }
-#endif
+  if (res != TRI_ERROR_NO_ERROR) {
+    // handle errors
+    if (errno == ENOMEM) {
+      // ENOMEM: The indicated memory (or part of it) was not mapped.
 
-  if (res == TRI_ERROR_NO_ERROR) {
-    // msync was successful
-    LOG_TOPIC(TRACE, Logger::MMAP) << "msync succeeded for range " << Logger::RANGE(startingAddress, numOfBytesToFlush) << ", file-descriptor " << fileDescriptor;
-    return TRI_ERROR_NO_ERROR;
-  }
+      // we have synced a region that was not mapped
+      // set a special error. ENOMEM (out of memory) is not appropriate
+      LOG_TOPIC(ERR, Logger::MMAP) << "msync failed for range " << Logger::RANGE(startingAddress, numOfBytesToFlush) << ", file-descriptor " << fileDescriptor << ": memory was not mapped";
 
-  if (errno == ENOMEM) {
-    // we have synced a region that was not mapped
+      return TRI_ERROR_ARANGO_MSYNC_FAILED;
+    }
 
-    // set a special error. ENOMEM (out of memory) is not appropriate
-    LOG_TOPIC(ERR, Logger::MMAP) << "msync failed for range " << Logger::RANGE(startingAddress, numOfBytesToFlush) << ", file-descriptor " << fileDescriptor;
+    if (errno == EINVAL) {
+      // EINVAL: addr is not a multiple of PAGESIZE; or any bit other than 
+      //         MS_ASYNC | MS_INVALIDATE | MS_SYNC is set in flags; or both
+      LOG_TOPIC(ERR, Logger::MMAP) << "msync failed for range " << Logger::RANGE(startingAddress, numOfBytesToFlush) << ", file-descriptor " << fileDescriptor << ": memory address or flags are invalid";
 
+      return TRI_ERROR_ARANGO_MSYNC_FAILED;
+    }
+    
```

Comment:
Should we add a case for `EBUSY`? Linux manpage suggests we can get it if "`MS_INVALIDATE` was specified in flags, and a memory lock exists for the specified address range."